### ISSUE TITLE
Fails to install via pipenv

### ISSUE THREAD

**Issue by ciaranevans** [pokepetter/ursina#169]
Hi there,

I'm having trouble trying to install `ursina==3.5.0` via `pipenv`

Whilst I realise the instructions say use `pip`, `pipenv` is an industry standard solution for dependency management and virtual env handling for Python, [PyPa even recommend to use it](https://packaging.python.org/tutorials/managing-dependencies/#managing-application-dependencies)

The error I'm receiving is:

```bash
$ pipenv install --dev
Pipfile.lock (69ecbb) out of date, updating to (0843df)...
Locking [dev-packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success! 
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✘ Locking Failed! 
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/resolver.py", line 741, in _main
[ResolutionFailure]:       resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages, dev)
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/resolver.py", line 702, in resolve_packages
[ResolutionFailure]:       results, resolver = resolve(
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/resolver.py", line 684, in resolve
[ResolutionFailure]:       return resolve_deps(
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/utils.py", line 1395, in resolve_deps
[ResolutionFailure]:       results, hashes, markers_lookup, resolver, skipped = actually_resolve_deps(
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/utils.py", line 1108, in actually_resolve_deps
[ResolutionFailure]:       resolver.resolve()
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/utils.py", line 833, in resolve
[ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: Could not find a version that matches panda3d (from ursina==3.5.0->-r /var/folders/v7/l6w8wpbs7g5d3q3w6p4_c8dw0000gn/T/pipenvtn2bjglcrequirements/pipenv-jzpprukh-constraints.txt (line 2))
No versions found
Was https://pypi.org/simple reachable?
```

For reference I am using:

* Python `3.8.8`
* Pipenv `2020.11.15`
* MacOS `11.2.3 (20D91)`

From what I understand of the error is that it's struggling to install `panda3d`. Could the fact that Ursina is not pinning its dependency versions be the issue?

I've also tried to specifically install `panda3d==1.10.9` and `panda3d-gltf==0.12` and receive a similar error:

```bash
$ pipenv install "panda3d==1.10.9" "panda3d-gltf==0.12" "ursina==3.5.0"
Installing panda3d==1.10.9...
Adding panda3d to Pipfile's [packages]...
✔ Installation Succeeded 
Installing panda3d-gltf==0.12...
Adding panda3d-gltf to Pipfile's [packages]...
✔ Installation Succeeded 
Installing ursina==3.5.0...
Adding ursina to Pipfile's [packages]...
✔ Installation Succeeded 
Pipfile.lock (69ecbb) out of date, updating to (35115c)...
Locking [dev-packages] dependencies...
Building requirements...
Resolving dependencies...
✔ Success! 
Locking [packages] dependencies...
Building requirements...
Resolving dependencies...
✘ Locking Failed! 
CRITICAL:pipenv.patched.notpip._internal.index.package_finder:Could not find a version that satisfies the requirement panda3d==1.10.9 (from -r /var/folders/v7/l6w8wpbs7g5d3q3w6p4_c8dw0000gn/T/pipenvyozbg_b6requirements/pipenv-hjxf5b_f-constraints.txt (line 2)) (from versions: none)
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/resolver.py", line 741, in _main
[ResolutionFailure]:       resolve_packages(pre, clear, verbose, system, write, requirements_dir, packages, dev)
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/resolver.py", line 702, in resolve_packages
[ResolutionFailure]:       results, resolver = resolve(
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/resolver.py", line 684, in resolve
[ResolutionFailure]:       return resolve_deps(
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/utils.py", line 1395, in resolve_deps
[ResolutionFailure]:       results, hashes, markers_lookup, resolver, skipped = actually_resolve_deps(
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/utils.py", line 1108, in actually_resolve_deps
[ResolutionFailure]:       resolver.resolve()
[ResolutionFailure]:   File "/usr/local/Cellar/pipenv/2020.11.15/libexec/lib/python3.9/site-packages/pipenv/utils.py", line 833, in resolve
[ResolutionFailure]:       raise ResolutionFailure(message=str(e))
[pipenv.exceptions.ResolutionFailure]: Warning: Your dependencies could not be resolved. You likely have a mismatch in your sub-dependencies.
  First try clearing your dependency cache with $ pipenv lock --clear, then try the original command again.
 Alternatively, you can use $ pipenv install --skip-lock to bypass this mechanism, then run $ pipenv graph to inspect the situation.
  Hint: try $ pipenv lock --pre if it is a pre-release dependency.
ERROR: No matching distribution found for panda3d==1.10.9 (from -r /var/folders/v7/l6w8wpbs7g5d3q3w6p4_c8dw0000gn/T/pipenvyozbg_b6requirements/pipenv-hjxf5b_f-constraints.txt (line 2))
```

I've followed all the instructions `pipenv` provides to try and get this to work but it doesn't. So I'm now coming here assuming it's something to do with how Ursina either pins its versions or is distributed.

**Comment by ciaranevans** [comment]
If I try to install `ursina==3.2.2` (The only tagged release) from [here](https://github.com/pokepetter/ursina/releases/tag/v3.2.2) I also get the same errors

**Comment by ciaranevans** [comment]
So I went to `panda3d`'s repository and raised an issue here https://github.com/panda3d/panda3d/issues/1145 regarding not being able to install `panda3d` on its own. This has now been resolved.

However I still could not install `ursina==3.5.0` even with the fixes applied from that issue.

I've found that if I specifically install `panda3d==1.10.9` separately and then install `ursina==3.5.0` then they both install OK. 

**TL;DR:**

I think Ursina needs to pin the versions of packages it uses in `setup.py`

**Comment by rdb** [comment]
I don't think Ursina should pin their versions just because of a bug in pipenv.  That would mean that Ursina does not get to benefit from the latest bug fixes in Panda3D, and users would be prevented from updating Panda3D if they run into such bugs.  At most you can do `panda3d~=1.10.9` but not `panda3d==1.10.9`.

**Comment by ciaranevans** [comment]
Closing as this is a `pipenv` issue, not `ursina`.

Many thanks @rdb for your detailed and quick help!
